<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="autoexposure" value="false"/>
    <param name="exposure" value="150"/>
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="io_method" value="mmap"/>
    <param name="camera_name" value="usb_cam" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="camera_info_url" value="file://$(find usb_cam)/calibration/usb_cam.yaml" />
  </node>

  <node pkg="aruco_ros" type="get_marker" name="aruco_single" output="screen">
    <param name="image_is_rectified" value="True"/>
    <param name="marker_size"        value="0.13"/>
    <param name="hz"                 value="30"/>
  </node>

  <node name="result" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/aruco_single/result"/>
    <param name="autosize" value="true" />
  </node>

  <node name="debug" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/aruco_single/debug"/>
    <param name="autosize" value="true" />
  </node>
</launch>

